unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/WorldChatWindowHUD/ChannelMembersHUDController.cs

```diff
@@ -0,0 +1,214 @@
+        {
+            ClearListeners();
+            view.Dispose();
+            loadingCancellationToken.Cancel();
+            loadingCancellationToken.Dispose();
+            reloadingCancellationToken.Cancel();
+            reloadingCancellationToken.Dispose();
+        }
+
+        public void SetVisibility(bool visible)
+        {
+            isVisible = visible;
+
+            if (visible)
+            {
+                LoadMembers();
+                SetAutomaticReloadingActive(true);
+            }
+            else
+            {
+                SetAutomaticReloadingActive(false);
+                ClearListeners();
+                view.Hide();
+            }
+        }
+
+        private void LoadMembers()
+        {
+            ClearListeners();
+
+            view.ClearSearchInput();
+            isSearching = false;
+
+            view.OnSearchUpdated += SearchMembers;
+            view.OnRequestMoreMembers += LoadMoreMembers;
+            chatController.OnUpdateChannelMembers += UpdateChannelMembers;
+
+            view.Show();
+            view.ClearAllEntries();
+            view.ShowLoading();
+
+            loadStartedTimestamp = DateTime.Now;
+            string[] channelsToGetInfo = { currentChannelId };
+            chatController.GetChannelInfo(channelsToGetInfo);
+            chatController.GetChannelMembers(currentChannelId, lastLimitRequested, 0);
+
+            loadingCancellationToken.Cancel();
+            loadingCancellationToken = new CancellationTokenSource();
+            WaitTimeoutThenHideLoading(loadingCancellationToken.Token).Forget();
+        }
+
+        private void SearchMembers(string searchText)
+        {
+            loadStartedTimestamp = DateTime.Now;
+            view.ClearAllEntries();
+            view.HideLoadingMore();
+            view.ShowLoading();
+
+            isSearching = !string.IsNullOrEmpty(searchText);
+
+            if (!isSearching)
+            {
+                chatController.GetChannelMembers(currentChannelId, lastLimitRequested, 0);
+                SetAutomaticReloadingActive(true);
+                view.HideResultsHeader();
+            }
+            else
+            {
+                chatController.GetChannelMembers(currentChannelId, LOAD_PAGE_SIZE, 0, searchText);
+                SetAutomaticReloadingActive(false);
+                view.ShowResultsHeader();
+            }
+
+            loadingCancellationToken.Cancel();
+            loadingCancellationToken = new CancellationTokenSource();
+            WaitTimeoutThenHideLoading(loadingCancellationToken.Token).Forget();
+        }
+
+        private void UpdateChannelMembers(string channelId, ChannelMember[] channelMembers)
+        {
+            if (!view.IsActive) return;
+            view.HideLoading();
+
+            foreach (ChannelMember member in channelMembers)
+            {
+                UserProfile memberProfile = UserProfileController.GetProfileByUserId(member.userId);
```

Comment:
avoid static calls, inject `IUserProfileBridge` instead

Reply:
You're right, I forgot inject it. Done!